src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java

```diff
@@ -0,0 +1,178 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
+
+/**
+ * <p>
+ * Checks if unnecessary semicolon is used after code type member declaration.
+ * The check will flag the following with warnings:
+ * </p>
+ * <pre>
+ * class A {
+ *     ; // standalone semicolon
+ *     {}; // semicolon after init block
+ *     static {}; // semicolon after static init block
+ *     A(){}; // semicolon after constructor definition
+ *     void method() {}; // semicolon after method definition
+ *     int field = 10;; // semicolon after field declaration
+ * }
+ * </pre>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;UnnecessarySemicolonAfterTypeMemberDeclaration&quot;/&gt;
+ * </pre>
+ *
+ * @since 8.24
+ */
+@StatelessCheck
+public final class UnnecessarySemicolonAfterTypeMemberDeclarationCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_SEMI = "unnecessary.semicolon";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
```

Comment:
@strkkk Is there a technical reason all tokens are required and can't be turned on/off?
@romani @pbludov We don't want to give users any possibility to turn off validation for specific tokens?

Reply:
I don't think that any of them can be optional. All of them look superfluous and erroneous.

Reply:
I do not see cases when someone want to have unnecessary semicolons, so no need to soften rules.

Reply:
Hmmm ... Let's not have required tokens. Users requirements are weird sometimes, let's not limit them. By default all tokens can are in use.

I had conversations at the past with users who run  Checks on generated code and wanted some flexibility.

In this Check case, no required tokens will give some freedom to user who want custom behavior ant this cost nothing for us.
But almost all users will use it in default form.

It is ok to skip testing of Check on non default set of tokens, as it was noticed a bit late.

Reply:
@strkkk , ping.

Reply:
removed